core/deployment/src/main/java/io/quarkus/deployment/dev/RuntimeUpdatesProcessor.java

```diff
@@ -94,6 +105,8 @@ public Path getClassesDir() {
         for (DevModeContext.ModuleInfo i : context.getAllModules()) {
             if (i.getResourcePath() != null) {
                 ret.add(Paths.get(i.getResourcePath()));
+            } else if (i.getResourcesOutputPath() != null) {
+                ret.add(Paths.get(i.getResourcesOutputPath()));
             }
         }
```

Comment:
@stuartwdouglas I'm thinking if checking the output path first is more adequate. On applications with filtered resources the handler for static resources provides the original and non-filtered files instead the filtered ones in output path. I've [asked on Zulip](https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Static.20resources.20filtering.20in.20dev.20mode) about that issue with filtered resources but no one replied yet.

I'll appreciate if you have a solution for that situation when proving filtered resources with static handler.

Reply:
I've answered on zulip too but just for completeness - open issue with reproducer of the problem is almost always to be preferred. 

So please go ahead and link to those places you found related to it.

Reply:
Thanks @maxandersen. I've created the issue #15303.